src/utils/bytes.js

```diff
@@ -38,6 +37,21 @@ export function utf8Decode(bytes) {
   });
 }
 
+// TODO(aghassemi, #6139): Remove the async version of utf8 encoding and rename
+// the sync versions to the canonical utf8Decode/utf8Encode.
+/**
+ * Interpret a byte array as a UTF-8 string.
+ * @param {!BufferSource} bytes
+ * @return {!string}
+ */
+export function utf8DecodeSync(bytes) {
+  if (typeof TextDecoder !== 'undefined') {
+    return new TextDecoder('utf-8').decode(bytes);
+  }
+  const asciiString = bytesToString(new Uint8Array(bytes));
```

Comment:
Excellent catch. Done. I think `ArrayBuffer|Uint8Array` makes much more sense for this, I don't see why allowing any `ArrayBufferView` other than `Uint8Array` is useful. MDN says `decode` takes `ArrayBufferView` which is even worse than what Closure Compiler says ( `BufferSource=ArrayBuffer|ArrayBufferView`).